core/src/main/java/org/apache/accumulo/core/Constants.java

```diff
@@ -99,7 +99,8 @@
   public static final long SCANNER_DEFAULT_READAHEAD_THRESHOLD = 3L;
 
   // Security configuration
-  public static final String PW_HASH_ALGORITHM = "SHA-256";
+  public static final String PW_HASH_ALGORITHM = "SHA-512";
```

Comment:
We still need this constant updated because we rely on it for hashing the system credentials? won't that prevent a rolling upgrade?

Could we have system credentials fall back to SHA-256 with a warning? or require a configurable flag to switch it?

Reply:
changing this is also going to change some non-security uses, e.g. we optionally use it to obscure values printed from rfile metrics gathering. We'll need to enumerate these and release note the change in behavior. (or we could make something like a `NON_CRYPTO_USE_HASH_ALGORITHM` that we keep as SHA-256)

Reply:
You are right. I'll make the system credentials hash customizable while defaulting to 256,  and renaming this as suggested to reflect it's for unsecure use.